lib/ash/actions/create/bulk.ex

```diff
@@ -221,11 +224,18 @@ defmodule Ash.Actions.Create.Bulk do
             opts[:return_notifications?] ->
               notifications
 
-            opts[:notify?] ->
-              Ash.Notifier.notify(notifications)
-              nil
-
             true ->
+              if opts[:notify?] do
+                Ash.Notifier.notify(notifications)
+              end
+
+              if Process.get(:ash_started_transaction?) do
+                Process.put(
+                  :ash_notifications,
+                  Process.get(:ash_notifications, []) ++ notifications
```

Comment:
I think we might want something like:

```elixir
notifications =
              if opts[:notify?] do
                 Ash.Notifier.notify(notifications)
               else
                 notifications
               end
```

in the section above so that we don't send some notifications and then also return them.